apps/designer/app/routes/designer/$id.tsx

```diff
@@ -37,47 +31,21 @@ type Error = {
 
 export const action: ActionFunction = async ({ request, params }) => {
   if (params.id === undefined) throw new Error("Project id undefined");
-  const uploads = path.join(__dirname, "../public");
-  const folderInPublic =
-    process.env.FILE_UPLOAD_PATH || config.defaultUploadPath;
-  const directory = path.join(uploads, folderInPublic);
-  try {
-    const formData = await unstable_parseMultipartFormData(
-      request,
-      unstable_createFileUploadHandler({
-        maxPartSize: 10_000_000,
-        directory,
-        file: ({ filename }) => filename,
-      })
-    );
-
-    const imagesInfo = ImagesUpload.parse(formData.getAll("image"));
-
-    const allInfo = imagesInfo.map(async (image) => {
-      const arrayBuffer = await image.arrayBuffer();
-      const data = {
-        name: image.name,
-        path: path.join("/", folderInPublic, image.name),
-        size: image.size,
-        arrayBuffer,
-      };
-
-      const projectId = params.id as string;
-      const newAsset = await db.assets.create(projectId, data);
-
-      return newAsset;
-    });
-
-    await Promise.all(allInfo);
-
-    return {
-      ok: true,
-    };
-  } catch (error) {
-    if (error instanceof Error) {
+  if (request.method === "POST") {
+    try {
+      const assets = await uploadAssets({
+        request,
+        projectId: params.id,
+        db,
+        dirname: __dirname,
+      });
       return {
-        errors: error.message,
+        assets,
       };
+    } catch (error) {
```

Comment:
you are missing the point of using try/catch, its used to handle the errors, but you just rethrow which is the same as not handling them

handling in this case is returning a response that is parsed by the client to show user an error, that's why we used in other cases `return {errors: 'something'}`